### DISCUSSION TITLE
Use custom fonts

### DISCUSSION THREAD

**Discussion by GrimPixel** [IndiePython/nodezator#56] [Feature Request]
The default font doesn't support many writing systems. The user may need to choose a system font.

**Comment by KennedyRichard** [comment]
Hello, Mr. GrimPixel,

Yep, this is a great feature request. To be more precise, the way Nodezator handles fonts needs some redesign to provide more customization to users.

This is important but it'll take some time to land due to other higher-priority features. I'll also need to finish implementing automated GUI tests before tackling this in order to define how to allow more customization without harming the automated GUI tests feature, though I believe there will be no significant problems. Handling IME will also be needed.

This is a fair request. I accept it :heavy_check_mark: 

My only addition is this: it should not apply to Python identifiers in the node, that is, the name of functions, parameters and outputs. It should apply only to data typed by the users (text in widgets and in comments/text boxes). This is so because according to [PEP8](https://peps.python.org/pep-0008/):

> All identifiers in the Python standard library MUST use ASCII-only identifiers, and SHOULD use English words wherever feasible

I think this is a reasonable and healthy expectation that facilitates cooperation between contributors from different nationalities and make things simpler in general. It is a good standard to adopt. In other words, the interface should allow people to handle data in any language desired, using the font of their choice. The names of functions, parameters, outputs and variables though should all use the default font provided by Nodezator and use ASCII-only characters. Anything else can be in any language the user desires, including the data they type and the interface of the app (menus, labels, panels, manual, etc.)

Finally, I'd just like to add that, of course, we must be careful not to let such standards become a barrier for non-English speakers to access technology and end up deepening social inequalities. I'm aware of that. It is just that I think it is best to encourage people (and when possible give support) to learn an international widespread language like English for the sake of easier cooperation and access to knowledge. As a native of a non-English speaking country, I see it for myself everyday and it saddens me how many opportunities are denied or not available to people because they can't understand/speak English.

Peace.

**Reply by GrimPixel** [reply]
I am glad that you are going to implement this. You are considerate, telling me the reason behind your decision. There is another discussion on the language: https://github.com/IndiePython/nodezator/discussions/54. 
On the one hand, languages are to be treated with egality; on the other hand, there needs to be a vehicular language. People need to be encouraged to learn languages, at least English in the current time, to learn about the world. 

I think it's time to introduce myself to you.
https://codeberg.org/GrimPixel
My current works are three, to cover knowledge from primary school to college: 
* SciTech Index: term linked with its prerequisites;
* Public Diagrams: diagrams under public licenses;
* Nodezator Equationary: node pack of equations.

The latter two are just started and will be published soon.

**Reply by KennedyRichard** [reply]
I just replied to the other discussion you mentioned. I read it yesterday but wanted to think a bit more about it.

Your work looks very interesting, can't wait to see your next steps considering your plans! Thank you for sharing, I'll keep an eye on it from time to time. It is a pleasure to have you in this humble community and to discuss your ideas, some of which are already listed as planned features.

Keep up the good work.

**Comment by KennedyRichard** [comment]
Added it to the list of planned/requested features on #13
